lib/orbit/transformable.js

```diff
@@ -1,121 +1,118 @@
 import Orbit from './main';
 import Evented from './evented';
 import ActionQueue from './action-queue';
-import Transformation from './transformation';
-import Operation from './operation';
-import { isArray } from './lib/objects';
+import Transform from './transform';
+import TransformResult from './transform-result';
 import { assert } from './lib/assert';
+import { deprecate } from './lib/deprecate';
 
-function normalize(operation) {
-  if (isArray(operation)) {
-    return operation.map(function(o) {
-      return normalize(o);
-    });
-
-  } else {
-    if (operation instanceof Operation) {
-      return operation;
-    } else {
-      return new Operation(operation);
-    }
-  }
+function settleTransform(transform, result) {
+  var _this = this;
+
+  this._transformLog[transform.id] = true;
+
+  return this.settle.call(this, 'didTransform', transform, result);
```

Comment:
I know we had discussed removing the Transaction class, but I haven't done so yet. Emitting the `result` here allows for inverse change tracking per source. We'll have to experiment to see how well change tracking could work if it's entirely internal to a source.